### PR TITLE
[998] Add ZAP Security Scan back into pipeline

### PR DESCRIPTION
**Pull request by DerekRoberts**

### CODE REVIEW

**Comment by DerekRoberts** [comment]
[EPIC#Pipeline | Pipeline + ZAP Tests |3| !!!](https://trello.com/c/P0dmQO7F/998-epicpipeline-pipeline-zap-tests-3)

File: Jenkinsfile [hunk 1]
```diff
@@ -245,6 +246,55 @@ _stage('Code Quality', context) {
...
+_stage('ZAP Security Scan', context) {
+    podTemplate(
+        label: 'owasp-zap${context.uuid}',
+        name: 'owasp-zap${context.uuid}',
+        serviceAccount: 'jenkins',
+        cloud: 'openshift',
+        containers: [
+            containerTemplate(
+                name: 'jnlp',
+                image: '172.50.0.2:5000/openshift/jenkins-slave-zap',
```
Comment (cvarjao) [inline Jenkinsfile]: Use FQDN: `docker-registry.default.svc:5000/openshift/jenkins-slave-zap`

File: Jenkinsfile [hunk 2]
```diff
@@ -245,6 +246,55 @@ _stage('Code Quality', context) {
...
+            echo "checking out source"
+            echo "Build: ${BUILD_ID}"
+            checkout scm
+            dir('zap') {
+                def retVal = sh (
+                    script: """
+                        set -eux
+                        ./runzap.sh
+                    """,
+                    returnStatus: true
```
Comment (cvarjao) [inline Jenkinsfile]: Notice that the exit value is not being analyzing and properly throwing exception when an error occurred (`reval != 0`):
see [](https://jenkins.io/doc/pipeline/steps/workflow-durable-task-step/#code-sh-code-shell-script)

File: Jenkinsfile [hunk 3]
```diff
@@ -245,6 +246,54 @@ _stage('Code Quality', context) {
 } //end stage
 
 
+_stage('ZAP Security Scan', context) {
+    podTemplate(
+        label: 'owasp-zap${context.uuid}',
```
Comment (cvarjao) [inline Jenkinsfile]: needs to use double quotes for proper string  interpolation.

File: Jenkinsfile [hunk 4]
```diff
@@ -245,6 +246,54 @@ _stage('Code Quality', context) {
...
+        name: 'owasp-zap${context.uuid}',
```
Comment (cvarjao) [inline Jenkinsfile]: needs to use double quotes for proper string  interpolation.

File: Jenkinsfile [hunk 5]
```diff
@@ -245,6 +246,54 @@ _stage('Code Quality', context) {
...
+                resourceLimitCpu: '1000m',
+                resourceRequestMemory: '3Gi',
+                resourceLimitMemory: '4Gi',
+                workingDir: '/home/jenkins',
+                command: '',
+                args: '${computer.jnlpmac} ${computer.name}'
+            )
+        ]
+    ) {
+        node('owasp-zap') {
```
Comment (cvarjao) [inline Jenkinsfile]: need to match the podTemplane name/label: "owasp-zap${context.uuid}"

**Pull request closed without merging**